contracts-periphery/test/UniswapWithdrawHook.fork.1.t.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "test/utils/DeployUmbraTest.sol";
+import "src/UniswapWithdrawHook.sol";
+
+contract UniswapWithdrawHookTest is DeployUmbraTest {
+  using SafeERC20 for IERC20;
+  UniswapWithdrawHook withdrawHook;
+
+  IUmbra umbraContract;
+  ISwapRouter swapRouter;
+  IQuoter quoterContract;
+  IERC20 daiToken;
+
+  uint256 toll;
+  address recipient;
+  address feeRecipient = address(0x20220613);
+  uint256 numOfAddrs = 10;
+  address sponsor;
+  uint256 quote;
+  uint256 feeAmount;
+
+  // Mainnet Addresses
+  address public constant dai = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+  address public constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  address public constant router = 0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45;
+  address public constant quoter = 0xb27308f9F90D607463bb33eA1BeBb41C27CE5AB6;
+  // 0.3% pool fee is unique to this DAI-WETH pool
+  uint24 poolFee = 3000;
+
+  // Umbra functions that we'll be calling
+  enum UmbraFns {withdrawTokenAndCall, withdrawTokenAndCallOnBehalf}
+
+  function setUp() public override {
+    super.setUp();
+    umbraContract = IUmbra(umbra);
+    swapRouter = ISwapRouter(router);
+    quoterContract = IQuoter(quoter);
+    withdrawHook = new UniswapWithdrawHook(ISwapRouter(swapRouter));
+    daiToken = IERC20(dai);
+    // Alice's address is derived from the private key of 1.
+    alice = vm.addr(1);
+    // Owner approves tokens
+    withdrawHook.approveToken(daiToken);
+    deal(dai, address(this), 10e29);
+  }
+
+  function testFuzz_HookTest_withdrawTokenAndCall(
+    uint256 amount,
+    uint256 swapAmount,
+    uint256 feeBips,
+    uint256 sponsorFee
+  ) public {
+    // Using for loop here to generate deterministic addresses instead of fetching random address state
+    // from network RPC node every time a fuzz test is run
+    for (uint256 i = 0; i < numOfAddrs; i++) {
+      feeRecipient = address(uint160(uint256(keccak256(abi.encode(i)))));
+      recipient = address(uint160(uint256(keccak256(abi.encode(feeRecipient)))));
+      _testFuzz_HookTest(
+        recipient,
+        amount,
+        swapAmount,
+        feeBips,
+        feeRecipient,
+        sponsorFee,
+        UmbraFns.withdrawTokenAndCall
+      );
+    }
+  }
+
+  function testFuzz_HookTest_withdrawTokenAndCallOnBehalf(
```

Comment:
This and `testFuzz_HookTest_withdrawTokenAndCall` both have the same `for` loop logic then call `_testFuzz_HookTest`. Maybe we dedupe that loop/logic by moving it into `_testFuzz_HookTest` instead?

Reply:
To do that won't we need to either not fuzz over vars like `amount`, `swapAmount`, and `feeBips` OR test both `testFuzz_HookTest_withdrawTokenAndCallOnBehalf` and `testFuzz_HookTest_withdrawTokenAndCall` within the same `_testFuzz_HookTest`?

Reply:
I think this current approach is fine, and you don't necessarily need to dedupe because it is a small piece of code / it's a test. But I'm not sure I understand the question, because all of those are already inputs to the `hookTest` method?

Reply:
I think the values passed in would end up being the same for every address we test, instead of being different like it is now.

Reply:
Hm why is that? Right now:
- Fuzzer generates values for amount, swapAmount, feeBips, and sponsorFee
- In a loop:
    - We generate values for the addresses
    - Pass the fuzzer data plus addresses to `hookTest` which executes the test in that loop

Possible but not required change:
- Fuzzer generates values for amount, swapAmount, feeBips, and sponsorFee
- That is passed to `hookTest`
- In a loop:
    - We generate values for the addresses
    - Then execute `hookTest` in that loop